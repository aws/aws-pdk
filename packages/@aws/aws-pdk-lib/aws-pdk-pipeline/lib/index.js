"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./pdk-pipeline"), exports);
__exportStar(require("./pdk-pipeline-ts-project"), exports);
__exportStar(require("./pdk-pipeline-py-project"), exports);
__exportStar(require("./pdk-pipeline-java-project"), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7O0FBQUEsaURBQStCO0FBQy9CLDREQUEwQztBQUMxQyw0REFBMEM7QUFDMUMsOERBQTRDIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lXCI7XG5leHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmUtdHMtcHJvamVjdFwiO1xuZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lLXB5LXByb2plY3RcIjtcbmV4cG9ydCAqIGZyb20gXCIuL3Bkay1waXBlbGluZS1qYXZhLXByb2plY3RcIjtcbiJdfQ==