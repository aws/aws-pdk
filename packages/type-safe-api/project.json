{
  "name": "@aws/type-safe-api",
  "root": "packages/type-safe-api",
  "targets": {
    "build": {
      "inputs": [
        "default",
        "^default",
        "!{projectRoot}/.jsii",
        "!{projectRoot}/lib/**/*",
        "!{projectRoot}/dist/**/*",
        "!{projectRoot}/coverage/**/*",
        "!{projectRoot}/test-reports/**/*",
        "!{projectRoot}/docs/api/**/*"
      ],
      "outputs": [
        "{projectRoot}/.jsii",
        "{projectRoot}/lib",
        "{projectRoot}/dist",
        "{projectRoot}/coverage",
        "{projectRoot}/test-reports",
        "{projectRoot}/docs/api"
      ],
      "dependsOn": [
        "^build"
      ],
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen build",
        "cwd": "packages/type-safe-api"
      }
    },
    "default": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen default",
        "cwd": "packages/type-safe-api"
      }
    },
    "pre-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen pre-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "post-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test",
        "cwd": "packages/type-safe-api"
      }
    },
    "package": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package",
        "cwd": "packages/type-safe-api"
      }
    },
    "test:watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test:watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint",
        "cwd": "packages/type-safe-api"
      }
    },
    "compat": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compat",
        "cwd": "packages/type-safe-api"
      }
    },
    "package-all": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package-all",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:java": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:java",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:python": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:python",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint-staged": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint-staged",
        "cwd": "packages/type-safe-api"
      }
    },
    "jest": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen jest",
        "cwd": "packages/type-safe-api"
      }
    },
    "docgen": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen docgen",
        "cwd": "packages/type-safe-api"
      }
    }
  },
  "implicitDependencies": [
    "smithy-async-transformer"
  ],
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"pnpm exec projen\"."
}
