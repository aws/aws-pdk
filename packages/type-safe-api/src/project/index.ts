/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./model/type-safe-api-model-build";
export * from "./model/type-safe-api-async-model-build";
export * from "./model/smithy/smithy-project-definition";
export * from "./model/openapi/open-api-project-definition";
export * from "./type-safe-api-project";
export * from "./model/type-safe-api-model-project";
export * from "./model/smithy/smithy-definition";
export * from "./model/openapi/open-api-definition";
export * from "./type-safe-websocket-api-project";
export * from "./model/type-safe-websocket-api-model-project";
export * from "./model/smithy/smithy-async-definition";
export * from "./model/openapi/open-api-async-definition";
export * from "./model/smithy/types";
export * from "./model/openapi/open-api-model-project";
export * from "./model/openapi/open-api-async-model-project";
export * from "./model/smithy/smithy-async-model-project";
export * from "./model/smithy/smithy-model-project";
export * from "./languages";
export * from "./types";
export * from "./typescript-project-options";
export * from "./python-project-options";
export * from "./java-project-options";
