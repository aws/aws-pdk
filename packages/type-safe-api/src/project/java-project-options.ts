// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
import { github, GitOptions, IgnoreFileOptions, java, LoggerOptions, Project, ProjectType, ProjenrcJsonOptions, RenovatebotOptions, SampleReadmeProps } from 'projen';

/**
 * JavaProjectOptions
 */
export interface JavaProjectOptions {
  /**
   * The java package to use for the code sample.
   * @default "org.acme"
   * @stability experimental
   */
  readonly sampleJavaPackage?: string;
  /**
   * Include sample code and test if the relevant directories don't exist.
   * @default true
   * @stability experimental
   */
  readonly sample?: boolean;
  /**
   * List of test dependencies for this project.
   * Dependencies use the format: `<groupId>/<artifactId>@<semver>`
   *
   * Additional dependencies can be added via `project.addTestDependency()`.
   * @default []
   * @stability experimental
   * @featured true
   */
  readonly testDeps?: Array<string>;
  /**
   * Options related to projenrc in java.
   * @default - default options
   * @stability experimental
   */
  readonly projenrcJavaOptions?: java.ProjenrcOptions;
  /**
   * Use projenrc in java.
   * This will install `projen` as a java dependency and will add a `synth` task which
   * will compile & execute `main()` from `src/main/java/projenrc.java`.
   * @default true
   * @stability experimental
   */
  readonly projenrcJava?: boolean;
  /**
   * Packaging options.
   * @default - defaults
   * @stability experimental
   */
  readonly packagingOptions?: java.MavenPackagingOptions;
  /**
   * junit options.
   * @default - defaults
   * @stability experimental
   */
  readonly junitOptions?: java.JunitOptions;
  /**
   * Include junit tests.
   * @default true
   * @stability experimental
   */
  readonly junit?: boolean;
  /**
   * Final artifact output directory.
   * @default "dist/java"
   * @stability experimental
   */
  readonly distdir?: string;
  /**
   * List of runtime dependencies for this project.
   * Dependencies use the format: `<groupId>/<artifactId>@<semver>`
   *
   * Additional dependencies can be added via `project.addDependency()`.
   * @default []
   * @stability experimental
   * @featured true
   */
  readonly deps?: Array<string>;
  /**
   * Compile options.
   * @default - defaults
   * @stability experimental
   */
  readonly compileOptions?: java.MavenCompileOptions;
  /**
   * The URL, like the name, is not required.
   * This is a nice gesture for
   * projects users, however, so that they know where the project lives.
   * @default undefined
   * @stability experimental
   * @featured true
   */
  readonly url?: string;
  /**
   * Project packaging format.
   * @default "jar"
   * @stability experimental
   */
  readonly packaging?: string;
  /**
   * Description of a project is always good.
   * Although this should not replace
   * formal documentation, a quick comment to any readers of the POM is always
   * helpful.
   * @default undefined
   * @stability experimental
   * @featured true
   */
  readonly description?: string;
  /**
   * This is the last piece of the naming puzzle.
   * groupId:artifactId denotes a
   * single project but they cannot delineate which incarnation of that project
   * we are talking about. Do we want the junit:junit of 2018 (version 4.12), or
   * of 2007 (version 3.8.2)? In short: code changes, those changes should be
   * versioned, and this element keeps those versions in line. It is also used
   * within an artifact's repository to separate versions from each other.
   * my-project version 1.0 files live in the directory structure
   * $M2_REPO/org/codehaus/mojo/my-project/1.0.
   * @default "0.1.0"
   * @stability experimental
   * @featured true
   */
  readonly version?: string;
  /**
   * This is generally unique amongst an organization or a project.
   * For example,
   * all core Maven artifacts do (well, should) live under the groupId
   * org.apache.maven. Group ID's do not necessarily use the dot notation, for
   * example, the junit project. Note that the dot-notated groupId does not have
   * to correspond to the package structure that the project contains. It is,
   * however, a good practice to follow. When stored within a repository, the
   * group acts much like the Java packaging structure does in an operating
   * system. The dots are replaced by OS specific directory separators (such as
   * '/' in Unix) which becomes a relative directory structure from the base
   * repository. In the example given, the org.codehaus.mojo group lives within
   * the directory $M2_REPO/org/codehaus/mojo.
   * @default "org.acme"
   * @stability experimental
   * @featured true
   */
  readonly groupId?: string;
  /**
   * The artifactId is generally the name that the project is known by.
   * Although
   * the groupId is important, people within the group will rarely mention the
   * groupId in discussion (they are often all be the same ID, such as the
   * MojoHaus project groupId: org.codehaus.mojo). It, along with the groupId,
   * creates a key that separates this project from every other project in the
   * world (at least, it should :) ). Along with the groupId, the artifactId
   * fully defines the artifact's living quarters within the repository. In the
   * case of the above project, my-project lives in
   * $M2_REPO/org/codehaus/mojo/my-project.
   * @default "my-app"
   * @stability experimental
   * @featured true
   */
  readonly artifactId?: string;
  /**
   * Enable VSCode integration.
   * Enabled by default for root projects. Disabled for non-root projects.
   * @default true
   * @stability experimental
   */
  readonly vscode?: boolean;
  /**
   * Auto-close stale issues and pull requests.
   * To disable set `stale` to `false`.
   * @default - see defaults in `StaleOptions`
   * @stability experimental
   */
  readonly staleOptions?: github.StaleOptions;
  /**
   * Auto-close of stale issues and pull request.
   * See `staleOptions` for options.
   * @default false
   * @stability experimental
   */
  readonly stale?: boolean;
  /**
   * The README setup.
   * @default - { filename: 'README.md', contents: '# replace this' }
   * @stability experimental
   */
  readonly readme?: SampleReadmeProps;
  /**
   * The name of a secret which includes a GitHub Personal Access Token to be used by projen workflows.
   * This token needs to have the `repo`, `workflows`
   * and `packages` scope.
   * @default "PROJEN_GITHUB_TOKEN"
   * @deprecated use `projenCredentials`
   * @stability deprecated
   */
  readonly projenTokenSecret?: string;
  /**
   * Choose a method of providing GitHub API access for projen workflows.
   * @default - use a personal access token named PROJEN_GITHUB_TOKEN
   * @stability experimental
   */
  readonly projenCredentials?: github.GithubCredentials;
  /**
   * Which type of project this is (library/app).
   * @default ProjectType.UNKNOWN
   * @deprecated no longer supported at the base project level
   * @stability deprecated
   */
  readonly projectType?: ProjectType;
  /**
   * Options for mergify.
   * @default - default options
   * @deprecated use `githubOptions.mergifyOptions` instead
   * @stability deprecated
   */
  readonly mergifyOptions?: github.MergifyOptions;
  /**
   * Whether mergify should be enabled on this repository or not.
   * @default true
   * @deprecated use `githubOptions.mergify` instead
   * @stability deprecated
   */
  readonly mergify?: boolean;
  /**
   * Add a Gitpod development environment.
   * @default false
   * @stability experimental
   */
  readonly gitpod?: boolean;
  /**
   * Options for GitHub integration.
   * @default - see GitHubOptions
   * @stability experimental
   */
  readonly githubOptions?: github.GitHubOptions;
  /**
   * Enable GitHub integration.
   * Enabled by default for root projects. Disabled for non-root projects.
   * @default true
   * @stability experimental
   */
  readonly github?: boolean;
  /**
   * Add a VSCode development environment (used for GitHub Codespaces).
   * @default false
   * @stability experimental
   */
  readonly devContainer?: boolean;
  /**
   * Add a `clobber` task which resets the repo to origin.
   * @default - true, but false for subprojects
   * @stability experimental
   */
  readonly clobber?: boolean;
  /**
   * Configure options for automatic merging on GitHub.
   * Has no effect if
   * `github.mergify` or `autoMerge` is set to false.
   * @default - see defaults in `AutoMergeOptions`
   * @stability experimental
   */
  readonly autoMergeOptions?: github.AutoMergeOptions;
  /**
   * Enable automatic merging on GitHub.
   * Has no effect if `github.mergify`
   * is set to false.
   * @default true
   * @stability experimental
   */
  readonly autoMerge?: boolean;
  /**
   * Enable and configure the 'auto approve' workflow.
   * @default - auto approve is disabled
   * @stability experimental
   */
  readonly autoApproveOptions?: github.AutoApproveOptions;
  /**
   * Options for renovatebot.
   * @default - default options
   * @stability experimental
   */
  readonly renovatebotOptions?: RenovatebotOptions;
  /**
   * Use renovatebot to handle dependency upgrades.
   * @default false
   * @stability experimental
   */
  readonly renovatebot?: boolean;
  /**
   * Options for .projenrc.json.
   * @default - default options
   * @stability experimental
   */
  readonly projenrcJsonOptions?: ProjenrcJsonOptions;
  /**
   * Generate (once) .projenrc.json (in JSON). Set to `false` in order to disable .projenrc.json generation.
   * @default false
   * @stability experimental
   */
  readonly projenrcJson?: boolean;
  /**
   * The shell command to use in order to run the projen CLI.
   * Can be used to customize in special environments.
   * @default "npx projen"
   * @stability experimental
   */
  readonly projenCommand?: string;
  /**
   * The parent project, if this project is part of a bigger project.
   * @stability experimental
   */
  readonly parent?: Project;
  /**
   * The root directory of the project.
   * Relative to this directory, all files are synthesized.
   *
   * If this project has a parent, this directory is relative to the parent
   * directory and it cannot be the same as the parent or any of it's other
   * sub-projects.
   * @default "."
   * @stability experimental
   */
  readonly outdir?: string;
  /**
   * Configure logging options such as verbosity.
   * @default {}
   * @stability experimental
   */
  readonly logging?: LoggerOptions;
  /**
   * Configuration options for git.
   * @stability experimental
   */
  readonly gitOptions?: GitOptions;
  /**
   * Configuration options for .gitignore file.
   * @stability experimental
   */
  readonly gitIgnoreOptions?: IgnoreFileOptions;
  /**
   * Whether to commit the managed files by default.
   * @default true
   * @stability experimental
   */
  readonly commitGenerated?: boolean;
  /**
   * This is the name of your project.
   * @default $BASEDIR
   * @stability experimental
   * @featured true
   */
  readonly name?: string;
}
