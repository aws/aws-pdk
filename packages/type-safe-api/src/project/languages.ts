/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
/**
 * Supported languages for runtimes and infrastructure
 */
export enum Language {
  TYPESCRIPT = "typescript",
  PYTHON = "python",
  JAVA = "java",
}

/**
 * Supported libraries for code generation
 */
export enum Library {
  /**
   * React query v4 hooks library
   * @deprecated Use react query v5 hooks library instead. This will be removed in PDK 1.0 release.
   */
  TYPESCRIPT_REACT_QUERY_HOOKS = "typescript-react-query-hooks",
  /**
   * React query v5 hooks library
   */
  TYPESCRIPT_REACT_QUERY_V5_HOOKS = "typescript-react-query-v5-hooks",
}

/**
 * Formats for documentation generation
 */
export enum DocumentationFormat {
  /**
   * HTML Documentation generated by redoc
   * @see https://github.com/Redocly/redoc
   */
  HTML_REDOC = "html_redoc",
  /**
   * OpenAPI Generator 'html2' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/html2.md
   */
  HTML2 = "html2",
  /**
   * OpenAPI Generator 'markdown' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/markdown.md
   */
  MARKDOWN = "markdown",
  /**
   * OpenAPI Generator 'plantuml' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/plantuml.md
   */
  PLANTUML = "plantuml",
}
