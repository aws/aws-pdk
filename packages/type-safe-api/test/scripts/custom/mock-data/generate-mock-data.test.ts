/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import os from "os";
import path from "path";
import { exec } from "projen/lib/util";
import { withTmpDirSnapshot } from "../../../project/snapshot-utils";

describe("Generate Mock Data Unit Tests", () => {
  it("Generates Mock Data", () => {
    expect(
      withTmpDirSnapshot(os.tmpdir(), (tmpDir) => {
        const specPath = "../../../resources/specs/single.yaml";
        const outputPath = path.relative(path.resolve(__dirname), tmpDir);
        const command = `../../../../scripts/type-safe-api/custom/mock-data/generate-mock-data --spec-path ${specPath} --output-path ${outputPath}`;
        exec(command, {
          cwd: path.resolve(__dirname),
        });
      })
    ).toMatchSnapshot();
  });
});
