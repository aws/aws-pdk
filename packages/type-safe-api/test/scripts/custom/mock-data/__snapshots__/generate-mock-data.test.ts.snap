// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Mock Data Unit Tests Generates Mock Data 1`] = `
{
  "mocks/delete-without-operation-id-200.json": {
    "messages": [
      {
        "id": 2359372120326144,
        "message": "canto eaque omnis",
      },
    ],
  },
  "mocks/get-map-response-200.json": {
    "mapProperty": {
      "voluptatibus-0": {
        "a": "cerno tabella cohors",
        "b": "ancilla thorax creptio",
      },
    },
  },
  "mocks/post-multiple-content-types-200.json": "allatus quisquam conventus",
  "mocks/post-path-{pathParam}-200.json": {
    "messages": [
      {
        "id": 2505140957347840,
        "message": "laborum articulus benevolentia",
      },
    ],
  },
  "mocks/post-path-{pathParam}-400.json": {
    "errorMessage": "illo degenero ademptio",
  },
  "mocks/put-any-request-response-200.json": {},
}
`;

exports[`Generate Mock Data Unit Tests Generates Mock Data For Many Data Types 1`] = `
{
  "mocks/get-types-200.json": {
    "myAdditionalProperties": {
      "theatrum-0": [
        14,
        17,
        10,
      ],
      "uberrime-1": [
        10,
        17,
        16,
      ],
    },
    "myAllOf": {
      "first": "arbustum benigne utor",
      "second": "decet aureus conculco",
    },
    "myAnyOf": "necessitatibus acsi candidus",
    "myBool": true,
    "myByte": "c3Blcm5vIGFlZ3JlIGFjcXVpcm8=",
    "myDateArray": [
      "2020-07-11",
    ],
    "myDateTime": "2021-09-20T00:25:04.928Z",
    "myEmail": "Gunner48@yahoo.com",
    "myHostname": "hopeful-pneumonia.org",
    "myInt": 4,
    "myIpv4": "115.87.166.36",
    "myIpv6": "ef26:0b5f:fc73:2969:a0f1:3f5c:4e4d:1feb",
    "myLongMinStringLength": "voluptatibus cerno tabella cohors ancilla thorax creptio allatus quisquam conventus ante talio vorago artificiose decipio unus testimonium texo thalassinus abscido contra delectus cupressus creator nulla eius temporibus corpus audeo demonstro civis urbanus spargo cultellus desidero decet atrox aqua cupiditate tracto avarus velit exercitationem vestrum tristis quaerat audax vita vita uxor currus torqueo desparatus stultus dolore suadeo accendo tui suus provident vulgo sono ascisco defungo antepono cupressus corrupti quos suadeo timidus abbas spiritus corporis ubi adsum temporibus testimonium vitium consectetur admoveo subvenio est deprecator spiritus voluptas tamquam tubineus fugit pel celo magni tum vulpes tabesco caterva undique accedo cognomen alienus peior vigor charisma voluptate vacuus deprecator civis confugo dolor supra arca abstergo sto temporibus valetudo speciosus fugiat aufero tener sophismata degero vere sopor arx itaque verbum texo communis utilis subseco universe cena cohaero earum vomer atrocitas supra delibero degenero deduco spectaculum crustulum vae textor denuncio depromo cupio surculus adsum textus studio consequuntur abundans verbera corpus summisse officiis velit denego curia balbus delibero somnus ventito volup recusandae deleo expedita accusantium capio celo vindico eveniet laboriosam arceo templum despecto canto coerceo curtus crapula cruentus vulariter dolorem theatrum antea bonus statua laudantium acsi sollicito canis sui tabgo error victus comprehendo placeat ager cumque corona ultra abundans vulgivagus quod clementia strues cribro tandem subnecto cado consuasor brevis addo cribro conscendo deporto deorsum aveho administratio confido veritas usus surgo demergo ter aranea alo vir adfero surgo caveo clarus antea abbas capio conduco statim tyrannus denique comptus video deficio nobis caelestis audax tres audentia voco confido ademptio damnatio bos aestus certus repellat aegrus adsum degero supra conor solutio complectus absorbeo thymum tenax ullus confido tabella tergeo maiores condico thalassinus nisi aspicio astrum considero tyrannus valens cerno pauci tam tersus magnam quas velut velum infit avaritia acies triduana aggero maxime vis paens defleo triumphus arceo aliquid spero subiungo quia cum inventore velum ventosus sono velociter tabernus celebrer adicio super terebro deorsum tabesco cicuta spiritus adsum cursim bardus enim rerum adhuc circumvenio veniam decipio vitiosus coniuratio cerno desidero vivo sollers decor vir apud ademptio dolorem iure suffoco saepe comitatus articulus cerno sursum ver cras pauci doloremque demergo nesciunt strenuus tutis depromo subvenio tutamen quo peior admoneo creta adipisci vigilo eveniet cervus terra cursus succedo talio auxilium via cernuus coaegresco bibo quidem cumque ademptio sollers provident cerno sumo cicuta vitiosus terminatio articulus ulterius quisquam incidunt paulatim surculus undique damnatio tum tracto pecto tredecim una acceptus est tabella adinventitias sunt atavus super substantia terror defleo viduo administratio solus cedo audeo subito sto suffoco vulariter quaerat vergo alter uredo abutor adipiscor suadeo absens cornu tamquam tonsor tabella verus suggero adeptio aro auditor angustus usque arto ventito trans curis culpo tantum atrox averto quos arto adficio cimentarius alius aestus amoveo nulla strenuus clibanus provident conicio natus temptatio trepide damnatio tamdiu absorbeo calamitas sperno alias cognatus desino corrumpo combibo cruciamentum ceno conduco dolor cauda substantia acceptus venustas sit defetiscor omnis expedita earum tolero cubitum creta vulpes talus ut vorax cura bene cunae nesciunt subiungo suppellex derelinquo appello ver torqueo annus tabula depopulo tam rerum turpis sed tandem unde talio reiciendis sapiente tibi rerum adsuesco tribuo caelum trucido strues brevis timor somniculosus eos attero solium trado ventus addo deficio adeo carpo accusantium nihil similique officiis exercitationem caput acsi ipsam acquiro cum tametsi sed tamisium ratione campana colligo terreo adnuo tutamen conqueror triduana communis itaque beatae stultus suus ter demens tactus quo cum vallum speciosus infit subiungo occaecati tot comparo conscendo adflicto error bestia aureus audentia casus id hic corrigo cogito iusto similique cursim administratio tabesco amo adeo templum corrupti decipio adulatio volubilis stillicidium annus sub contabesco trepide vesica cariosus tubineus repellat volutabrum atrox autus confugo contego cenaculum similique accommodo charisma claustrum adipiscor spargo nobis cognatus audeo deleo testimonium crebro crastinus absens compello caterva soleo tres quibusdam spoliatio color sponte tamisium auxilium culpa vinculum teneo aequitas adinventitias casso verbera via sint comptus defessus cohibeo acerbitas acquiro bellicus caterva placeat validus vestrum in peccatus articulus laudantium dolor cupio conduco tubineus tracto nobis aperte tam beneficium cariosus labore amplus comparo vox impedit adduco adipiscor brevis cernuus cuppedia ascit capto nisi cena adipisci iure auctus pel umquam pectus templum temptatio caelestis bibo deorsum varietas absque solutio volup adipisci adeptio patria depono crapula valeo coniecto taceo avarus earum sumo facere voluptatum caste chirographum verumtamen sollers labore debilito sub damnatio volutabrum amiculum sponte caste dolor ustilo bellum ancilla versus comes aqua sumo eos crastinus vestrum cerno convoco vindico conturbo verbum coerceo pel crastinus vulnus denique vos ancilla utrimque adeo desino tunc tondeo dignissimos theca corrigo atrocitas mollitia pectus adimpleo alter curiositas calco desino arbitro ultio collum ager defessus auditor clibanus corpus laboriosam surculus ut correptius beatae thorax adsuesco atrocitas ver tepesco conforto distinctio turba caterva vicissitudo confido tolero tutamen advenio suspendo culpa est cauda autus sequi verto nam decerno vespillo vergo cruentus quaerat tui ventus statim vereor concido cubitum ciminatio caste sufficio usitas molestias id volup talio bonus clamo canto caritas suasoria coerceo solvo acervus verto suppono vesco temperantia casus arcesso fuga rerum crustulum consectetur aliqua confido certe adiuvo ait incidunt voluptatibus recusandae velum tandem triumphus odit volva video patria crinis sponte summopere brevis ustulo vester adfero adficio stips ratione corrupti adflicto pauper voluptate taedium utique advoco desparatus sol odit tergiversatio cicuta aliquid argumentum demergo tricesimus tabgo apostolus earum apparatus tardus coniecto desidero voveo possimus denuncio statua cohors vigor carus crapula terminatio dolorum arca uterque decens vulgo ustilo vito abutor officia templum conturbo volva casus candidus dolorem taedium explicabo thesaurus tego ut tabernus subito currus bos temporibus aggredior adulescens laboriosam repellendus laboriosam beneficium cubo adfero in thymbra aspernatur cum atavus velit sumo atrox vulnus error thesis ago asperiores comes teres crastinus clamo conitor vestigium aro demergo crapula cubicularis armarium alveus timidus despecto trucido alias suggero stillicidium vehemens apto cimentarius delinquo tamquam carmen nisi xiphias pecto certe virga numquam adflicto accedo antea perferendis carus dolorem commodi denuncio compello culpa expedita valetudo nesciunt auctus ater suadeo callide tergiversatio considero tabernus voluntarius pectus molestiae subseco admoveo caput valde ait vereor verbum spoliatio facilis repellendus carbo cuius turpis benevolentia totam corrigo capitulus vapulus uter adduco",
    "myNot": "quidem advoco utpote",
    "myNotString": {},
    "myNumber": 0.5658850050531328,
    "myOneOf": "victus sto agnosco",
    "myRegexPattern": "5555-pattern-nnn",
    "myString": "laborum articulus benevolentia",
    "myStringLength": "ademp",
    "myUrl": "https://incompatible-diner.info/",
    "myUuid": "19106449-d1d7-4db3-a05e-9b0ad4a56244",
  },
}
`;

exports[`Generate Mock Data Unit Tests Generates Mock Data For Recursive Definitions 1`] = `
{
  "mocks/get-tree-200.json": {},
}
`;

exports[`Generate Mock Data Unit Tests Generates Mock Data For Recursive Definitions With Required Recursive Reference 1`] = `
{
  "mocks/get-tree-200.json": {
    "next": {
      "next": {},
      "number": 2.620246761944145,
    },
    "number": 5.605297528672963,
  },
}
`;
